synapseclient/client.py

```diff
@@ -400,15 +403,35 @@ def __init__(
             cache.CACHE_ROOT_DIR if cache_root_dir is None else cache_root_dir
         )
 
-        config_debug = None
         # Check for a config file
+        config_debug = None
         self.configPath = configPath
-        if os.path.isfile(configPath):
-            config = get_config_file(configPath)
-            if config.has_option("cache", "location"):
-                cache_root_dir = config.get("cache", "location")
-            if config.has_section("debug"):
-                config_debug = True
+        if os.path.isfile(self.configPath):
+            config = get_config_file(self.configPath)
+
+            self.auth_profiles = {}
+            for section in config.sections():
+                if section.startswith("profile "):
+                    profile_name = section.split("profile ")[-1]
+                    self.auth_profiles[profile_name] = {
+                        "username": config.get(section, "username", fallback=None),
+                        "authtoken": config.get(section, "authtoken", fallback=None),
+                    }
+                elif section == "authentication":
+                    self.auth_profiles["default"] = {
+                        "username": config.get(section, "username", fallback=None),
+                        "authtoken": config.get(section, "authtoken", fallback=None),
+                    }
+        # If no profiles exist, create an empty one
+        if not hasattr(self, "auth_profiles") or not self.auth_profiles:
+            self.auth_profiles = {"default": {"username": None, "authtoken": None}}
+
+            '''OKAy bottom is og code, above to me whatever CM is edited 
+            '''
+            # if config.has_option("cache", "location"):
+            #     cache_root_dir = config.get("cache", "location")
+            # if config.has_section("debug"):
+            #     config_debug = True
```

Comment:
I don't think this code is necessary is it? When a user calls the `syn.login(profile="user1")` method it performs a read of the given config file at that point in time. I t seems like in this case we do not need to do anything different in the `Synapse.__init__` function.